script/prepare-VAFs.py

```diff
@@ -0,0 +1,233 @@
+    for (tool, vcf_path, purity) in opts.vcf_data:
+        validate_tool(tool)
+        validate_purity(purity)
+        validate_vcf(vcf_path)
+
+    # If a tool is repeated, the latest value will be used
+    # Not erroring out in this case in case we want to support
+    # multiple tools/samples in the future
+
+    validate_output_dir(opts.output_dir)
+
+def get_vaf_strelka2(variant: dict, sample: str) -> Tuple[int, int]:
+    """ Calculate read count for Strelka2 format """
+    sample_index = variant.samples.index(sample)
+    ref_base = variant.REF.strip(' ') + 'U'
+    ref_reads = int(variant.samples[sample_index][ref_base][0])
+    # pylint: disable=R1728
+    # Take allele with most number of reads as the ALT
+    alt_reads = max([int(variant.samples[sample_index][allele.sequence + 'U'][0]) \
+        for allele in variant.ALT])
+    total_reads = ref_reads + alt_reads
+    return alt_reads/total_reads
+
+def get_vaf_somaticsniper(variant: dict, sample: str) -> Tuple[int, int]:
+    """ Calculate read count for SomaticSniper format """
+    # using DP4
+    sample_index = variant.samples.index(sample)
+    ref_reads = int(variant.samples[sample_index]["DP4"][0]) \
+        + int(variant.samples[sample_index]["DP4"][1])
+    alt_reads = int(variant.samples[sample_index]["DP4"][2]) \
+        + int(variant.samples[sample_index]["DP4"][3])
+    total_reads = ref_reads + alt_reads
+    return alt_reads/total_reads
+
+def get_vaf_mutect2(variant: dict, sample: str) -> Tuple[int, int]:
+    """ Calculate read count for Mutect2 format """
+    sample_index = variant.samples.index(sample)
+    ref_reads = int(variant.samples[sample_index]["AD"][0])
+    alt_reads = int(variant.samples[sample_index]["AD"][1])
+    total_reads = ref_reads + alt_reads
+
+    return alt_reads/total_reads
+
+def does_variant_pass(variant: dict) -> bool:
+    """ Does variant pass filter checks """
+    return (variant.FILTER is None or (len(variant.FILTER) == 0))
+
+def calculate_adjusted_vaf(sample_id: str, variant: dict, purity: float) -> float:
+    """ Determine function to calculate VAF and return adjusted VAF """
+    sample = None
+    for a_sample in variant.samples:
+        if a_sample.sample == sample_id:
+            sample = a_sample
+            break
+
+    if not sample:
+        raise ValueError(f"{sample_id} not found.")
+
+    sample_index = variant.samples.index(sample)
+    variant_data_keys = variant.samples[sample_index].data._asdict().keys()
+
+    raw_vaf = 0
+
+    if 'AD' in variant_data_keys:
+        raw_vaf = get_vaf_mutect2(variant, sample)
+    elif 'DP4' in variant_data_keys:
+        raw_vaf = get_vaf_somaticsniper(variant, sample)
+    elif all(base+'U' in variant_data_keys for base in ['A', 'C', 'G', 'T']):
+        raw_vaf = get_vaf_strelka2(variant, sample)
+    else:
+        raise ValueError(
+            f'Unable to identify variant source for: {variant} with {variant_data_keys}'
+        )
+
+    return raw_vaf * purity
```

Comment:
Am i correct that the VAF for any particular variant is extracted from any one of the input VCFs (based on order here)?   The different tools don't all provide the same numbers, `Mutect2` seems to provide filtered read count, where the others provide unfiltered or both (`Strelka2`).   Is this accounted for in any way?  I'm not sure there is a simple way.

Reply:
This is correct, the VAF for all variants is extracted in a tool-specific manner, ie. there is a VAF extracted per variant per tool and then averaging steps are done to reconcile/combine the VAFs. So it's accounted for through averaging essentially